File: backend/core/events.ts
export const auth = {
    signIn: 'auth:sign-in',
    current: 'auth:current-user',
    updatedUser: 'auth:user-update',
    error: 'auth:error',
    locked: 'auth:locked',
    unlocked: 'auth:unlocked'
}


export const tables = {
    room: 'tables-room',
    // Requests
    create: 'tables:create',
    modify: 'tables:modify',
    delete: 'tables:delete',
    initialize: 'tables:initialize',

    // events
    createdTableEvent: 'tables:created-table',
    updatedTableEvent: 'tables:updated-table',
    deletedTableEvent: 'tables:deleted-table',

    //Responses
    initialData: 'tables:initial-data',
    successCreated: 'tables:success-created',
    succesDeleted: 'tables:success-deleted',
    succesModifiedd: 'tables:success-modified',

    //Errors
    errorCreated: 'tables:error-created',
    errorDeleted: 'tables:error-deleted',
    errorModified: 'tables:error-modified'

}


export const products = {
    room: 'products-room',
    // Requests
    create: 'products:create',
    modify: 'products:modify',
    delete: 'products:delete',
    initialize: 'products:initialize',

    // events
    createdProductEvent: 'products:created-table',
    updatedProductEvent: 'products:updated-table',
    deletedProductEvent: 'products:deleted-table',

    //Responses
    initialData: 'products:initial-data',
    successCreated: 'products:success-created',
    succesDeleted: 'products:success-deleted',
    succesModifiedd: 'products:success-modified',

    //Errors
    errorCreated: 'products:error-created',
    errorDeleted: 'products:error-deleted',
    errorModified: 'products:error-modified'
}


export const categories = {
    room: 'categories-room',
    // Requests
    create: 'categories:create',
    modify: 'categories:modify',
    delete: 'categories:delete',
    initialize: 'categories:initialize',

    // events
    createdTableEvent: 'categories:created-table',
    updatedTableEvent: 'categories:updated-table',
    deletedTableEvent: 'categories:deleted-table',

    //Responses
    initialData: 'categories:initial-data',
    successCreated: 'categories:success-created',
    succesDeleted: 'categories:success-deleted',
    succesModifiedd: 'categories:success-modified',

    //Errors
    errorCreated: 'categories:error-created',
    errorDeleted: 'categories:error-deleted',
    errorModified: 'categories:error-modified'
}

File: kairos/lib/models/events.dart
class EventSet {
  // requests
   final String create;
   final String modify;
   final String delete;
   final String initialize;

  // server-events
   final String createdEvent;
   final String updatedEvent;
   final String deletedEvent;

  //Responses
   final String initialData;
   final String successCreated;
   final String successDeleted;
   final String successModified;

  //Errors
   final String initError;
   final String errorCreated;
   final String errorDeleted;
   final String errorModified;

  const EventSet(String prefix)
      : create = '$prefix:create',
        modify = '$prefix:modify',
        delete = '$prefix:delete',
        initialize = '$prefix:initialize',
        createdEvent = '$prefix:created-table',
        updatedEvent = '$prefix:updated-table',
        deletedEvent = '$prefix:deleted-table',
        initialData = '$prefix:initial-data',
        successCreated = '$prefix:success-created',
        successDeleted = '$prefix:success-deleted',
        successModified = '$prefix:success-modified',
        initError = '$prefix:init-error',
        errorCreated = '$prefix:error-created',
        errorDeleted = '$prefix:error-deleted',
        errorModified = '$prefix:error-modified';
}
